crates/rust-analyzer/src/handlers.rs

```diff
@@ -746,6 +746,19 @@ fn handle_fixes(
     let file_id = from_proto::file_id(&snap, &params.text_document.uri)?;
     let line_index = snap.analysis.file_line_index(file_id)?;
     let range = from_proto::text_range(&line_index, params.range);
+
+    match &params.context.only {
+        Some(v) => {
+            if v.iter().any(|it| {
+                it == &lsp_types::CodeActionKind::EMPTY
+                    || it == &lsp_types::CodeActionKind::QUICKFIX
+            }) {
+                return Ok(());
```

Comment:
Hm, why do we need this case?

Reply:
The check is wrong... it should be negated (we do not want to compute fixes if not requested)

Reply:
Fixed in the latest